src/ray/core_worker/task_manager.cc

```diff
@@ -186,6 +231,31 @@ void TaskManager::ShutdownIfNeeded() {
   }
 }
 
+void TaskManager::FailPendingActorTasks(const ActorID &actor_id) {
+  std::list<TaskID> actor_tasks;
+  {
+    absl::MutexLock lock(&mu_);
+    actor_tasks = std::move(pending_actor_tasks_[actor_id]);
+    pending_actor_tasks_.erase(actor_id);
+  }
+  for (auto& task_id : actor_tasks) {
+    PendingTaskFailed(task_id, rpc::ErrorType::ACTOR_DIED);
+  }
+}
+
+void TaskManager::ResubmitActorTasks(const ActorID &actor_id) {
```

Comment:
Btw, another thought I had is that it probably wouldn't work to just resubmit everything anyway, since some tasks may be buffered in the `CoreWorkerDirectActorSubmitter`, so we could end up with duplicate task submissions. Maybe you've already accounted for that, but just another thing to be aware of.

Reply:
Yeah I explicitly flush the submitter every time a connection is dropped (which didn't happen previously).